### PR TITLE
HFSS assign symmetry implementation

### PR DESCRIPTION
**Pull request by gmalinve**

### CODE REVIEW

**Review by maxcapodi78 — COMMENTED** [review]
Please check my comment

File: pyaedt/hfss.py [hunk 1]
```diff
@@ -5452,3 +5452,60 @@ def set_material_threshold(self, threshold=100000):
...
             return False
+
+    @pyaedt_function_handler()
+    def assign_symmetry(self, entity_list, symmetry_name=None, is_perfect_e=True):
+        """Assign symmetry to planar entities.
+
+        Parameters
+        ----------
+        entity_list : list
+            List IDs or :class:`pyaedt.modeler.Object3d.FacePrimitive`.
```
Comment (MaxJPRey) [inline pyaedt/hfss.py:5463]:
```suggestion
            List of IDs or :class:`pyaedt.modeler.Object3d.FacePrimitive`.
```

File: pyaedt/hfss.py [hunk 2]
```diff
@@ -5452,3 +5452,60 @@ def set_material_threshold(self, threshold=100000):
...
+        symmetry_name : str, optional
+            Name of the boundary.
+            If not provided it's automatically generated.
+        is_perfect_e : bool, optional
+            Type of symmetry plane the boundary represents: Perfect E or Perfect H.
+            Default value is `True` (Perfect E).
```
Comment (MaxJPRey) [inline pyaedt/hfss.py:5469]:
```suggestion
            The default value is ``True`` (Perfect E).
```

File: pyaedt/hfss.py [hunk 3]
```diff
@@ -5452,3 +5452,60 @@ def set_material_threshold(self, threshold=100000):
...
+        >>> ids = [i.id for i in hfss.modeler["SymmetryForFaces"].faces]
+        >>> symmetry = hfss.assign_symmetry(ids)
+        >>> type(symmetry)
+        <class 'pyaedt.modules.Boundary.BoundaryObject'>
+
+        """
+        try:
+            if self.solution_type != "Modal":
+                msg = "Assign symmetry cannot be performed on solution types different from Modal."
+                raise ValueError(msg)
```
Comment (MaxJPRey) [inline pyaedt/hfss.py:5497]:
```suggestion
                raise ValueError("Symmetry is only available with 'Modal' solution type.")
```

File: pyaedt/hfss.py [hunk 4]
```diff
@@ -5452,3 +5452,60 @@ def set_material_threshold(self, threshold=100000):
...
+
+            if symmetry_name is None:
+                symmetry_name = generate_unique_name("Symmetry")
+
+            if not isinstance(entity_list, list):
+                msg = "Entities have to be provided as a list."
+                raise ValueError(msg)
```
Comment (MaxJPRey) [inline pyaedt/hfss.py:5504]: @gmalinve Here, I am nitpicking as above. Usually we create a variable only if it is used more than once.
```suggestion
                raise ValueError("Entities have to be provided as a list.")
```

**Review by MaxJPRey — APPROVED** [review]
Looks great.

**Review by maxcapodi78 — APPROVED** [review]

**Pull request merged**